glovo/src/main/java/org/example/glovo/service/OrderService.java

```diff
@@ -0,0 +1,124 @@
+    }
+
+    public Order update(Order order) throws OrderNotFoundException {
+        if (order.getId() == null) {
+            throw new IllegalArgumentException("Order id is a required argument");
+        }
+
+        orderRepository.findById(order.getId())
+                .orElseThrow(() -> new OrderNotFoundException("Order not found"));
+
+        OrderEntity updateOrderEntity = orderRepository.save(OrderMapper.toEntity(order));
+
+        return OrderMapper.toModel(updateOrderEntity);
+    }
+
+    public Order getById(Long id) {
+        if (id == null) {
+            throw new IllegalArgumentException("Order not found");
+        }
+
+        OrderEntity foundOrder = orderRepository.findById(id)
+                .orElseThrow(() -> new IllegalArgumentException("Order not found"));
+
+        return OrderMapper.toModel(foundOrder);
+    }
+
+    public Order addProduct(Long id, Product product) throws OrderNotFoundException {
+        if (id == null) {
+            throw new IllegalArgumentException("Order id is a required argument");
+        }
+
+        OrderEntity foundOrder = orderRepository.findById(id)
+                .orElseThrow(() -> new OrderNotFoundException("Order not found"));
+
+        ProductEntity addedProductEntity = productService.add(product);
+        foundOrder.getProducts().add(addedProductEntity);
+
+        OrderEntity updateOrderEntity = orderRepository.save(foundOrder);
+
+        return OrderMapper.toModel(updateOrderEntity);
+    }
+
+    public Order addProduct(Long orderId, Long productId) throws OrderNotFoundException, ProductNotFoundException {
+        if (orderId == null) {
+            throw new IllegalArgumentException("Order id is a required argument");
+        }
+
+        OrderEntity foundOrder = orderRepository.findById(orderId)
+                .orElseThrow(() -> new OrderNotFoundException("Order not found"));
+
+        ProductEntity existingProductEntity = productService.getById(productId);
+        foundOrder.getProducts().add(existingProductEntity);
+
+        OrderEntity updatedOrderEntity = orderRepository.save(foundOrder);
+
+        return OrderMapper.toModel(updatedOrderEntity);
+    }
+
+    public boolean deleteProduct(Long orderId, Long productId) throws OrderNotFoundException, ProductNotFoundException {
+        if (orderId == null) {
+            throw new IllegalArgumentException("Order id is a required argument");
+        }
+
+        OrderEntity foundOrder = orderRepository.findById(orderId).orElseThrow(() -> new OrderNotFoundException("Order not found"));
+
+        ProductEntity existingProductEntity = productService.getById(productId);
+        foundOrder.getProducts().remove(existingProductEntity);
+
+        orderRepository.save(foundOrder);
+
+        return true;
+    }
+
+    public Order delete(Long id) throws OrderNotFoundException {
+        if (id == null) {
+            throw new IllegalArgumentException("Order id is a required argument");
+        }
```

Comment:
may be better to go with primitive  long instead of Long

Reply:
agree!